src/bin/ion/commands/beta/generate/templates/rust/anonymous_type.templ

```diff
@@ -0,0 +1,135 @@
+{# following macro defines an anonymous type as children class for its parent type definition #}
+{% macro anonymous_type(target_kind_name, fields, abstract_data_type, nested_anonymous_types) -%}
+    #[derive(Debug, Clone, Default)]
+    pub struct {{ target_kind_name }} {
+    {% for field in fields -%}
+             {{ field.name | snake | indent(first = true) }}: {{ field.value_type }},
+    {% endfor %}
+    }
+
+    impl {{ target_kind_name }} {
+        pub fn new({% for field in fields | sort(attribute="name") -%}{{ field.name | snake }}: {{ field.value_type }},{% endfor %}) -> Self {
+            Self {
+                {% for field in fields -%}
+                {{ field.name | snake }},
+                {% endfor %}
+            }
+        }
+
+
+        {% for field in fields -%}pub fn {{ field.name | snake }}(&self) -> &{{ field.value_type }} {
+            &self.{{ field.name | snake }}
+        }
+        {% endfor %}
+
+
+        pub fn read_from(reader: &mut Reader) -> SerdeResult<Self> {
+            let mut abstract_data_type = {{ target_kind_name }}::default();
+            {% if abstract_data_type == "Value"%}
+                abstract_data_type.value = {% if fields[0].value_type | is_built_in_type == false %}
+                                                {{ fields[0].value_type }}::read_from(reader)?;
+                                            {% else %}
+                                                reader.read_{% if fields[0].isl_type_name == "symbol" %}symbol()?.text().unwrap(){% else %}{{ fields[0].value_type | replace(from="string", to ="str") }}()?{% endif %}{% if fields[0].value_type | lower == "string" %} .to_string() {% endif %};
+                                            {% endif %}
+            {% elif abstract_data_type is object and abstract_data_type is containing("Structure") %}
+                reader.step_in()?;
+                while reader.next()? != StreamItem::Nothing {
+                    if let Some(field_name) = reader.field_name()?.text() {
+                        match field_name {
+                            {% for field in fields -%}
+                                {% if field.value_type | is_built_in_type == false %}
+                                     "{{ field.name }}" => { abstract_data_type.{{ field.name | snake }} = {{ field.value_type }}::read_from(reader)?; }
+                                {% else %}
+                                    "{{ field.name }}" => { abstract_data_type.{{ field.name | snake}} = reader.read_{% if field.isl_type_name == "symbol" %}symbol()?.text().unwrap(){% else %}{{ field.value_type | lower | replace(from="string", to ="str") }}()?{% endif %}{% if field.value_type | lower== "string" %} .to_string() {% endif %}; }
```

Comment:
Is there a `rustfmt` for Tera templates? I'm wondering if we can automate some of the line-wrap management.

Reply:
I tried to find some IDE plugin for templates, it does have some formatting for jinja templates but that would align all the statements in same indentation leading it difficult to read the normal Java/Rust statements in the template. I will try to see if I can find some better solution for this.